data/tutorials/getting-started/02_fst_program.md

```diff
@@ -0,0 +1,308 @@
+
+```shell
+$ cd hello
+$ tree
+.
+├── bin
+│   ├── dune
+│   └── main.ml
+├── _build
+│   └── log
+├── dune-project
+├── hello.opam
+├── lib
+│   └── dune
+└── test
+    ├── dune
+    └── hello.ml
+
+4 directories, 8 files
+```
+
+OCaml source files have `.ml` extension, which stands for “Meta Language.” Meta Language (ML) is the ancestor of OCaml, this is also what the “ml” stands for in “OCaml.” Here is the content of the `bin/main.ml` file:
+
+```shell
+$ cat ./bin/main.ml
+let () = print_endline "Hello, World!"
+```
+
+The project-wide metadata is available in the `dune-project` file. Each folder containing source files that need to be built must contain a `dune` file explaining how.
+
+This builds the project:
+```shell
+$ dune build
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+```
+
+This launches the executable it defines:
+```shell
+$ dune exec hello
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+Hello, World!
+```
+
+Edit the `bin/main.ml` file. Replace `World` by your first name. Recompile the project. Launch it againg. Here your are, this is your first OCaml program.
+
+In the rest of this tutorial, we will make changes to this project to illustrate OCaml tooling and 
+
+## Why no Main Function? 
+
+Although `bin/main.ml`'s name accurately suggests it contains the application entry point into the executable binary produced by the project, it does not contain a dedicated `main` function. Even that file name is irrelevant, it could be anything. This is because OCaml is equally an interpreted and compiled language. Once compiled, binaries are executed as if their source counterpart where read.
+
+That's why double semicolumns are no longer needed in source files, statements are just processed in order, from top to bottom, each triggering the side effects it may have. Definitons are added to the environment. Values resulting from nameless expressions are ignored. Side effects from all those will take place in the same order. That's OCaml main.
+
+However, it is common practice to single out a value that triggers all the side effects and mark it as the intended main entry point. In OCaml, that's the role of the `let () =` pattern, which stands for “definition without a name” and actually indicates we're only interested in the side effects of the value on the right.
+
+## Modules and the Standard Library, Cont'd
+
+Let's summarise what was said about modules in the previous tutorial:
+- Modules are bundles of named values 
+- Identical names from distinct modules don't clash
+- The standard library is a bunch of modules
+
+In addition to that, modules are also the mean of organising projects. Concerns are separated into isolated modules. This is outlined in the next section. Before that, let's see how to use a definition from a module in our project: 
+```shell
+$ sed -i 's/print_endline/Printf.printf "%s!\\n"/g' bin/main.ml
```

Comment:
Without the bin/main.ml I get this error: `sed: -I or -i may not be used with stdin`, and with the bin/main.ml in the command, I get this error: `sed: 1: "bin/main.ml": undefined label 'in/main.ml'`. I'm in the `hello` directory.

Reply:
(I've stopped testing the code this time through until I'm sure I'm not doing anything wrong to get these errors. I'll continue line editing."

Reply:
I get this same error, even with the updates.

Reply:
This doesn't work in macOS as-is, so Cuit will rewrite this (or at least speak to it, so the reader knows what to expect).